sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -133,8 +134,21 @@ case class StaticInvoke(
   override def nullable: Boolean = needNullCheck || returnNullable
   override def children: Seq[Expression] = arguments
 
-  override def eval(input: InternalRow): Any =
-    throw new UnsupportedOperationException("Only code-generated evaluation is supported.")
+  override def eval(input: InternalRow): Any = {
+    if (staticObject == null) {
+      throw new RuntimeException("The static class cannot be null.")
+    }
+
+    val parmTypes = arguments.map(e =>
+      CallMethodViaReflection.typeMapping.getOrElse(e.dataType,
+        Seq(e.dataType.asInstanceOf[ObjectType].cls))(0))
```

Comment:
The external types of native types `CalendarIntervalType` and `BinaryType` are not `ObjectType`.

Reply:
You are right. I have to support other types before merging this change.
This is a prototype for discussing whether we use reflection or not.